websites/A/AnimeHeaven/presence.ts

```diff
@@ -0,0 +1,143 @@
+const presence = new Presence({
+    clientId: "816042675626442783"
+  }),
+  strings = presence.getStrings({
+    play: "presence.playback.playing",
+    pause: "presence.playback.paused"
+  }),
+  browsingStamp = Math.floor(Date.now() / 1000);
+
+type VideoData = {
+  paused: boolean;
+  currentTime: number;
+  duration: number;
+};
+let iFrameVideo: VideoData;
+presence.on("iFrameData", (videoData: VideoData) => {
+  iFrameVideo = videoData;
+});
+
+const paths = {
+  async "/watch"(presenceData: PresenceData, buttons: boolean) {
+    const title = document.querySelector(".now2 .c a")?.textContent,
+      episode = document
+        .querySelector(".now2 .c")
+        ?.lastChild?.textContent?.match(/\s*-\s*(.+)/)[1];
+
+    presenceData.details = title || "Unknown title";
+    presenceData.state = episode || "Unknown episode";
+
+    if (buttons) {
+      const link = (document.querySelector(
+        ".now2 > div > a"
+      ) as HTMLLinkElement)?.href;
+      presenceData.buttons = [
+        {
+          label: "Current Episode",
+          url: window.location.href
+        },
+        link
+          ? {
+              label: "Episode List",
+              url: link
+            }
+          : undefined
```

Comment:
never set undefined/null to presenceData, if you want to assign buttons, assign them first to different variable and then delete undefined/null whatever.

Reply:
It makes no difference though?

Reply:
Why not? It's the same outcome, it's just written different